src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -104,17 +110,56 @@ protected override void OnNextCore(Block block)
             this.logger.LogTrace("(-)");
         }
 
+        /// <summary>Initializes the <see cref="BlockStoreSignaled"/>.</summary>
+        public void Initialize()
+        {
+            this.localBatch = new List<ChainedBlock>();
+
+            // Configure batch timer.
+            this.batchTimer = new Timer(this.batchInterval.TotalMilliseconds);
+            this.batchTimer.AutoReset = false;
+            this.batchTimer.Elapsed += async (sender, args) =>
+            {
+                using (await this.asyncLock.LockAsync(this.nodeLifetime.ApplicationStopping).ConfigureAwait(false))
+                {
+                    if (!this.batchTimer.Enabled)
+                        return;
+
+                    await SendBatchAsync();
```

Comment:
if this method needs lock to be called, it should have Locked suffix and that contract should be documented

Reply:
Fixed.